### DISCUSSION TITLE
Autotype with window focus change

### DISCUSSION THREAD

**Discussion by chainsawbeaver** [keepassxreboot/keepassxc#10898] [Q&A] [answered]
I try to get autotype to work on the Cisco Secure Client.
The sequence is {PASSWORD}{ENTER}{DELAY 1000}{TOTP}{ENTER}
This is where I start the sequence with the keyboard shortcut.
![2](https://github.com/keepassxreboot/keepassxc/assets/5499917/851ab1f8-1250-48df-be86-f4f75d1a1275)
The password is typed by keepassxc, Then new window pops up:
![3](https://github.com/keepassxreboot/keepassxc/assets/5499917/dab44452-aa73-4f95-bf85-a4a9278dee32)
and autotype stops working.

If I start autotype in Notepad++ the TOTP is typed.

I switched from keepass2 to keepassxc, and on keepass2 the full autotype sequence works.

Is this a bug, or a feature (because the windows focus changed), or do I miss something?  Thank you

**Comment by droidmonkey** [comment, marked as answer]
You cannot use auto-type across two different windows. You need to invoke a new auto-type action.

**Reply by RicardoCapricciosa** [reply]
Well, it works for me across the second window, apart from #12497 (see attached video there)

**Reply by chainsawbeaver** [reply]
perhaps in MacOS it is not a new window but a "sub"-window. I have no idea about the right terminology. In your video it looks like both windows are connected to the main application window. On Windows 11 these are to windows that pop up with no "parent" window. Does not help with your bug, sorry.

**Reply by RicardoCapricciosa** [reply]
Seems so, yes. I guess in your case a new feature would be required, where it's possible to combine different sequences for different windows into one sequence. A feature like this would be useful for me as well.
